controllers/suite_test.go

```diff
@@ -236,110 +138,73 @@ var _ = BeforeEach(func() {
 		Spec: corev1.PodSpec{
 			Containers: []corev1.Container{
 				{
-					Image: "bar",
+					Image: "k8s.gcr.io/pause:3.4.1",
 					Name:  "ctn1",
 				},
-				{
-					Image: "bar",
-					Name:  "ctn2",
-				},
-				{
-					Image: "bar",
-					Name:  "ctn3",
-				},
-			},
-		},
-	}
-	targetPodC = &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "car",
-			Namespace: "default",
-			Labels: map[string]string{
-				"foo": "bar",
-			},
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Image: "car",
-					Name:  "ctn1",
-				},
-				{
-					Image: "car",
-					Name:  "ctn2",
-				},
-				{
-					Image: "car",
-					Name:  "ctn3",
-				},
-			},
-		},
-	}
-	targetPodD = &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "far",
-			Namespace: "default",
-			Labels: map[string]string{
-				"foo": "bar",
-			},
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Image: "far",
-					Name:  "ctn1",
-				},
-				{
-					Image: "far",
-					Name:  "ctn2",
-				},
-				{
-					Image: "far",
-					Name:  "ctn3",
-				},
 			},
 		},
 	}
 
-	By("Creating target pod")
-	err := k8sClient.Create(context.Background(), targetPodA)
-	if apierrors.IsInvalid(err) {
-		logf.Log.Error(err, "failed to create object, got an invalid object error")
-		return
-	}
-	Expect(err).NotTo(HaveOccurred())
+	By("Creating target pods")
+	Expect(k8sClient.Create(context.Background(), targetPodA)).ToNot(HaveOccurred())
```

Comment:
I don't understand, what are we expecting to have not occurred here?

Reply:
nvm, it's the err returned by Create

Reply:
Yup this assertion is misleading. I think `.To(BeNil())` would have the same behavior and eventually be more explicit?

Reply:
Yes, that's what I'm used to from our tests, and I think is more explicit, but maybe that's because i'm more familiar with it? This (to me) reads like we expect the Create to have not occurred, even though it really means the error